libs/payments/cart/src/lib/cart.service.spec.ts

```diff
@@ -380,29 +380,46 @@ describe('CartService', () => {
   });
 
   describe('getCart', () => {
+    const mockPrice = StripePriceFactory();
+    const mockUpcomingInvoice = StripeResponseFactory(
+      StripeUpcomingInvoiceFactory({
+        total_discount_amounts: [
+          {
+            amount: 500,
+            discount: StripeDiscountFactory(),
+          },
+        ],
+        total_tax_amounts: [
+          {
+            amount: faker.number.int(1000),
+            inclusive: false,
+            tax_rate: StripeTaxRateFactory(),
+            taxability_reason: null,
+            taxable_amount: null,
+          },
+        ],
+      })
+    );
+
+    const mockInvoicePreview = {
+      currency: mockUpcomingInvoice.currency,
+      listAmount: mockUpcomingInvoice.amount_due,
+      totalAmount: mockUpcomingInvoice.total,
+      taxAmounts: [
+        {
+          title: mockUpcomingInvoice.total_tax_amounts[0].tax_rate.display_name,
+          inclusive: mockUpcomingInvoice.total_tax_amounts[0].inclusive,
+          amount: mockUpcomingInvoice.total_tax_amounts[0].amount,
+        },
+      ],
+      discountAmount:
+        mockUpcomingInvoice.total_discount_amounts &&
+        mockUpcomingInvoice.total_discount_amounts[0].amount,
```

Comment:
Not sure if it's any better, but I think you can also use the [optional chaining (`?.`) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to avoid the extra `&&` bit:

```suggestion
        mockUpcomingInvoice.total_discount_amounts?.[0].amount,
```